cmd/vm_edit.go

```diff
@@ -0,0 +1,70 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/exoscale/egoscale"
+	"github.com/spf13/cobra"
+)
+
+var vmEditCmd = &cobra.Command{
+	Use:   "edit <name|ID>",
+	Short: "Edit virtual machine properties",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var (
+			vmEdit egoscale.UpdateVirtualMachine
+			edited bool
+		)
+
+		if len(args) != 1 {
+			return cmd.Usage()
+		}
+
+		vm, err := getVirtualMachineByNameOrID(args[0])
+		if err != nil {
+			return err
+		}
+		vmEdit.ID = vm.ID
+
+		name, err := cmd.Flags().GetString("name")
+		if err != nil {
+			return err
+		}
+		if cmd.Flags().Changed("name") {
+			vmEdit.Name = name
+			vmEdit.DisplayName = name
+			edited = true
+		}
+
+		userDataPath, err := cmd.Flags().GetString("cloud-init-file")
+		if err != nil {
+			return err
+		}
+		if userDataPath != "" {
+			vmEdit.UserData, err = getUserDataFromFile(userDataPath)
+			if err != nil {
+				return err
+			}
+			edited = true
+		}
+
+		if edited {
```

Comment:
IMO I Don't think the `edited` bool is useful, I've tried with cs you can update an instance without parameters, only with ID it's works and update nothing.

Reply:
@pierre-emmanuelJ good to know, but in this case it also allows us to give a feedback to the user, signaling that the command *actually* changed something. Otherwise, we'd always return "Virtual machine updated successfully" even though we didn't do anything, which is confusing from a UX standpoint.

Reply:
I would also be in favour of not returning "Virtual machine updated successfully" if no flag is present.

I can as well observe the following behaviour in other commands:

    $ exo vm scale
    error: required flag(s) "service-offering" not set

    $ exo vm updateip
    Usage:
      exo vm updateip <vm name|id> <network name|id> [flags]
      ...
      
So i'd say that imho the best would be to have something like

    $ exo vm edit
    Usage:
      exo vm edit <vm name|id> [flags]
      ...

for non conform calls of the command.

Reply:
@ste-m fixed as suggested:

```console
$ ./bin/exo vm edit testvm
Usage:
  exo vm edit <name|ID> [flags]

Flags:
      --cloud-init-file string   path to a cloud-init user data file
  -h, --help                     help for edit
      --name string              display name

Global Flags:
  -C, --config string            Specify an alternate config file [env EXOSCALE_CONFIG]
  -O, --output-format string     Output format (table|json|text), see "exo output --help" for more information
      --output-template string   Template to use if output format is "text"
  -Q, --quiet                    Quiet mode (disable non-essential command output)
  -A, --use-account string       Account to use in config file [env EXOSCALE_ACCOUNT]
```

Reply:
@ste-m  as discussed earlier with @falzm in PM, would be great to return the `vm show` output  as result to see the changes, if some flags is set of course.

Since the call `getVirtualMachineUserData` is in the API maybe we can do something with that to add it in the output result. I think it can be also useful to support the discussion below with @exo-cedric and @falzm.